src/bot.ts

```diff
@@ -52,23 +54,28 @@ client.on('interactionCreate', async (i) => {
 		const command = botCmds.get(i.commandName);
 
 		if (command) {
-			const missingPermissions = getMissingPermissions(command.permissions, i);
+			if (i.guild) {
+				await setNewGuild(i.user.id, i.guild.id);
 
-			if (missingPermissions) {
-				i.reply(
-					t('missing_permissions', {
-						perms: missingPermissions.join(' '),
-					}),
+				const missingPermissions = checkPermissions(
+					command.permissions,
+					i.guild,
 				);
 
-				return;
-			}
+				if (missingPermissions) {
+					i.reply(
+						t('missing_permissions', {
+							perms: missingPermissions.join(' '),
+						}),
+					);
 
-			if (i.guildId) {
-				await setNewGuild(i.user.id, i.guildId);
-			}
+					return;
+				}
 
-			command.execute(i);
+				command.execute(i);
+			} else {
+				command.execute(i);
+			}
```

Comment:
```suggestion
			}
			command.execute(i);
```
Since you're calling it in both true and false, can remove it from the blocks.